### PR TITLE
fix(cloud): retain inference standing lifecycle and primary moderation

### PR DESCRIPTION
**Pull request by lalalune**
Remove a redundant awaited tier-cache write from cold inference admission and give the authorization resolver sole ownership of retained usage updates. After unban, invalidate cached API-key, session, and outbound denials; reject explicitly when any eviction cannot be confirmed; authoritative refresh reads moderation and identities from the primary so replica lag cannot immediately restore stale standing.

Refs #30695, #30696, #30700, #30706.

The staging capability diagnostic now records `healthObservedDeploySha` separately and explicitly marks the capability response deployment as unverified. Matching health checks around a HEAD request cannot identify the replica that served that request. Schema version 2 removes the misleading generic `deploySha`; a mixed-replica regression protects the distinction. The diagnostic remains HEAD-only and does not claim authentication success or permission to mutate accounts.

The original implementation and published history are retained after rebasing onto develop `63323d0f77f2b48eb6221fafbcb243ab3154b21d`. The generated inventory repair has landed in develop and is no longer part of this PR diff.

A false result from outbound standing deletion now raises `MODERATION_OUTBOUND_INVALIDATION_UNCONFIRMED` instead of reporting successful moderation. The regression uses production outbound standing and cache helpers with deterministic database and mock-Redis boundaries: a failed deletion leaves the denial observable, unban rejects, and retry after cache recovery succeeds and removes it. The test no longer mocks outbound invalidation as always successful.

Validation at `5336cdcf82c08751a0bb4e6fdc68ca21f7b0a3db` on develop `63323d0f77f2b48eb6221fafbcb243ab3154b21d`:

- Normal pinned `bun install` passed with no lockfile changes.
- `RUN_TURBO_CONCURRENCY=1 bun run verify` passed: 374/374 tasks, then every repository audit.
- Baseline complete Cloud shared suite at published `dd80b97` passed all 348 batches: 12,597 passed, 209 skipped, 0 failed. Skipped live lanes are not live-provider acceptance.
- Focused changed Cloud/script suites passed: 151 tests, 602 assertions. Capability boundary suite passed all 10 tests.
- Cloud shared typecheck and touched-file Biome passed for the correction. The previous root run with the same correction on base `40e6498` passed all 374 tasks and repository audits.
- Real local HTTP inspection served health revision B, capability revision C, then health revision B. The artifact retained B only as `healthObservedDeploySha` and marked `capabilityDeployment` unverified. No credentials or account identities were emitted.
- An earlier full-suite attempt overlapped a shared dist rebuild and failed two advertising imports. The subsequent sequential full run above passed; no code workaround or test exclusion was introduced.
- `git diff --check` passes. The existing published head remains an ancestor, so publication uses a normal fast-forward push.

Local HTTP evidence uses controlled mixed-revision responses, not staging. No live latency saving or production change is claimed.

Post-merge acceptance remains the protected staging capability check, an authorized disposable standing canary, and exact-deployment latency certification. Cache deletion propagation can remain eventual.

- [x] N/A - no rendered UI behavior changes.

- [x] N/A - no rendered UI behavior changes.

- [x] N/A - backend admission and a standalone diagnostic.

- [x] Pinned local commands and results are recorded above; baseline full-suite results and final-head regression results are distinguished above.

```text
$ RUN_TURBO_CONCURRENCY=1 bun run verify
 Tasks:    374 successful, 374 total
Cached:    0 cached, 374 total
  Time:    6m21.758s
Exit status: 0

$ bun test --isolate [10 changed Cloud/script suites]
 151 pass
 0 fail
 602 expect() calls
Ran 151 tests across 10 files. [3.63s]
Exit status: 0
```

- [x] N/A - no frontend behavior change.

- [x] N/A - no prompts or model output changes; post-merge live admission certification remains separately identified.

- [x] The complete controlled local HTTP observation is below.

<details>
<summary>Controlled mixed-replica HTTP observation</summary>

```json
{
  "harness": "real local HTTP with mixed revision responses; no staging request",
  "revisions": {
    "health": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
    "capability": "cccccccccccccccccccccccccccccccccccccccc"
  },
  "requests": [
    {
      "method": "GET",
      "path": "/api/health"
    },
    {
      "method": "HEAD",
      "path": "/api/v1/admin/moderation"
    },
    {
      "method": "GET",
      "path": "/api/health"
    }
  ],
  "evidence": {
    "schemaVersion": 2,
    "kind": "staging_standing_capability",
    "environment": "staging",
    "sourceSha": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "healthObservedDeploySha": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
    "capabilityDeployment": {
      "status": "unverified",
      "reason": "response_revision_not_verified"
    },
    "observedAt": "2026-09-06T18:39:55.828Z",
    "httpStatus": 200,
    "isAdmin": true,
    "role": "moderator",
    "provesAuthentication": false,
    "provesModerationMutation": false
  }
}
```
</details>

- [x] N/A - no visual artifacts.

### CODE REVIEW

**Review by ss251 — CHANGES_REQUESTED** [review]
03bf20ed55adbe140faaa6e691c62222f458515e

[P2] The capability artifact attributes the HEAD response to a deployment it has not identified. In `packages/cloud/scripts/staging-standing-capability.mjs:81-117`, deployment checks and the authenticated HEAD are three independent requests; `deploySha` is then copied from the configured expectation. With mixed staging replicas, health(A) → moderation HEAD(B) → health(A) passes and emits an observation labeled A even though the role headers came from B. This matters because the workflow publishes that JSON as deployment-specific evidence (`.github/workflows/staging-standing-capability.yml:57-63`).

Validate deployment identity on the capability response itself, using server-provided revision metadata, before assigning it `deploySha`. Alternatively, label the field as the revision observed only on the surrounding health requests and explicitly state that the capability response's revision is unverified. Add a mixed-revision boundary case; the existing deployment-movement test only changes the health responses (`packages/cloud/scripts/staging-standing-capability.test.mjs:100-110`).

The primary moderation reads and unban invalidation address a concrete stale-standing risk. However, the unban test manually writes the subsequent allowed cache entries rather than exercising an authorization refresh (`packages/cloud/shared/src/lib/services/admin-unban-invalidation.test.ts:197-225`); it establishes eviction and helper behavior, not end-to-end restored access.

VERIFICATION: Statically read the supplied diff at this head, including credential dispatch, evidence construction, and moderation/cache tests. Live execution blocked: no disposable sandbox on this host; no tests or staging requests executed.

Recommendation: fix
no commands: live execution blocked

AI provider/model: Experiential Labs / gpt-6-astra
Client / agent tooling: experiential-gateway
Contribution skill revision: SlopDotCash/slopdotcash@ffc07b0ca4888df20faee5b490181dbcaae5cc91:skills/contribute-to-eliza
Attribution status: self-reported
— [xpl-gpt-6-astra-lane-ss251]

**Comment by NubsCarson** [comment]
Re-reviewed the deployment-attribution correction at dd80b97bd39bfdfb879587291cd4949cbc0b9cea. The diagnostic now records only healthObservedDeploySha and explicitly marks the capability response deployment unverified; the generic deploySha is removed. I independently ran the exact-head standalone boundary suite: 10 tests passed, including mixed-replica attribution. This resolves the earlier scoped attribution finding. Current-head hosted checks are still running; this comment is not full PR approval or live standing/latency acceptance.

**Comment by lalalune** [comment]
I independently confirmed the outbound invalidation finding at dd80b97bd39bfdfb879587291cd4949cbc0b9cea: invalidateOutboundMessageStanding returns the confirmed-delete boolean, but invalidateUserInferenceContext discards it. A failed deletion can therefore report a completed unban while a cached denial still blocks outbound work. The current unban fixture returns true for this helper and does not exercise failed deletion. This needs an observed failure plus successful retry regression before merge. @NubsCarson I am coordinating this correction with the merge queue; please flag any in-progress fix so we avoid competing branch updates. I have not pushed over your head.

File: packages/cloud/shared/src/lib/services/admin.ts [hunk 1]
```diff
@@ -28,13 +28,13 @@ import { invalidateOutboundMessageStanding } from "./outbound-message-standing";
...
  */
 async function invalidateUserInferenceContext(userId: string): Promise<void> {
   const [keys, user] = await Promise.all([
-    apiKeysRepository.listByUser(userId),
-    dbRead.query.users.findFirst({
+    apiKeysRepository.listByUserConsistent(userId),
+    dbWrite.query.users.findFirst({
       where: eq(users.id, userId),
       columns: { steward_user_id: true, organization_id: true },
     }),
```
[the remark this answers is not in the document]
Comment (NubsCarson) [reply packages/cloud/shared/src/lib/services/admin.ts:41]: Fixed in 5336cdcf82c08751a0bb4e6fdc68ca21f7b0a3db. The moderation transition now rejects an unconfirmed outbound-standing delete with MODERATION_OUTBOUND_INVALIDATION_UNCONFIRMED. The regression exercises the real outbound/cache helper: failed eviction preserves the cached denial and rejects unban; retry after confirmed eviction clears it. Red/green verified, 151 focused tests and the full root verify gate pass on this head.

File: packages/cloud/shared/src/db/repositories/api-keys.ts [hunk 2]
```diff
@@ -179,6 +179,13 @@ export class ApiKeysRepository {
     });
   }
 
+  /** Lists every key on the primary before standing-cache invalidation. */
+  async listByUserConsistent(userId: string): Promise<ApiKey[]> {
+    return await dbWrite.query.apiKeys.findMany({
+      where: eq(apiKeys.user_id, userId),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (NubsCarson) [reply packages/cloud/shared/src/db/repositories/api-keys.ts:187]: The consistent variant intentionally preserves the existing listByUser repository return contract while changing its read authority. This is a moderation-transition path, not the inference-admission hot path; no measured latency claim is made for this lookup. A narrower projection is reasonable optimization work, but it is not required for the primary-read and fail-closed invalidation contract repaired here.

**Comment by lalalune** [comment]
Independent follow-up at 5336cdcf82c08751a0bb4e6fdc68ca21f7b0a3db: the published outbound guard resolves the confirmed false-delete issue; no competing implementation is needed. I ran the exact public moderation cache suite (7 passed, 34 assertions) and capability suite (10 passed). Source review confirms unban/clear_status cannot return success after this error, and the existing API boundary sanitizes it to HTTP 500. I am running the complete Cloud shared suite on this exact head to supplement the documented dd80 baseline; that run and hosted 34055713590 are still active. This is scoped review evidence, not yet terminal full-suite or deployed acceptance.
